### PR TITLE
General fixes for event & Bookmarked Event List View

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
### General Event Fixes

- Get rid of edit event screen and make all the details editable by inline actions
- Restrict user to update RSVP when event is past
- Restrict user to change dates when event is past/ongoing
- Minor UI fixes

Reference Video:

https://github.com/user-attachments/assets/b8802ff1-3229-4c77-8c8d-46289f02bc2d

Side-note:
- Even after using event description with generic edit description code, [this issue](https://github.com/acterglobal/a3-meta/issues/252) note get fixes.
Scenario :  Edit Event not working if there is no text set while creating event. Edit Description is working when there is some text already set and we try to change it.

Issue Root Cause : https://github.com/acterglobal/a3/pull/2044/files#r1705407701

https://github.com/user-attachments/assets/88ef6986-5830-432f-9cd3-d7b71123f683

---

### Bookmarked Event List
- On top of above fixes and with the continuation of https://github.com/acterglobal/a3/pull/2035, added bookmarked events list in [this commit](https://github.com/acterglobal/a3/pull/2044/commits/563ff88eb372c366a0d383168ce31bb39572f73b)

https://github.com/user-attachments/assets/b364986b-47af-41fb-be8b-18ddf0a54c9c

### CODE REVIEW

File: app/lib/features/events/pages/event_details_page.dart [hunk 1]
```diff
@@ -635,40 +666,20 @@ class _EventDetailPageConsumerState extends ConsumerState<EventDetailPage> {
...
   }
 
-  void showEditDescriptionSheet(String? formattedText, String? bodyText) {
+  void showEditDescriptionSheet(CalendarEvent ev) {
+    TextMessageContent? content = ev.description();
     showEditHtmlDescriptionBottomSheet(
       context: context,
-      descriptionHtmlValue: formattedText,
-      descriptionMarkdownValue: bodyText,
+      descriptionHtmlValue: content?.formatted(),
```
Comment (kumarpalsinh25) [inline app/lib/features/events/pages/event_details_page.dart:673]: `content?.formatted()`
This code is seems like not working correctly, It is causing issue on our HtmlEdtior.

File: app/lib/features/events/utils/events_utils.dart
Comment (gnunicorn) [inline app/lib/features/events/utils/events_utils.dart:1]: shouldn't this be in `app/lib/features/events/actions/`?

File: app/lib/features/events/widgets/change_date_sheet.dart [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+        _selectedEndDate = date;
+        _endDateController.text = eventDateFormat(date);
+        // When user change date that time end time is reset
+        _endTimeController.text = '';
+      } else {
+        EasyLoading.showToast(L10n.of(context).pleaseSelectValidEndDate);
+      }
+    }
+    setState(() {});
+  }
```
Comment (gnunicorn) [inline app/lib/features/events/widgets/change_date_sheet.dart:278]: nitpick: this sort of things screams for widget unittests to make sure they always work correctly and don't get broken upon later changes...

**Review by gnunicorn — APPROVED** [review]
Looking good. Just minor inquiries, to be done as you please.

File: app/lib/features/events/widgets/change_date_sheet.dart [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    _selectedStartDate = dartStartTime.toLocal();
+    _selectedStartTime = TimeOfDay.fromDateTime(_selectedStartDate);
+    _startDateController.text = eventDateFormat(_selectedStartDate);
+    _startTimeController.text = _selectedStartTime.format(context);
+
+    // Setting data to variables for end date
+    _selectedEndDate = dartEndTime.toLocal();
+    _selectedEndTime = TimeOfDay.fromDateTime(_selectedEndDate);
+    _endDateController.text = eventDateFormat(_selectedEndDate);
+    _endTimeController.text = _selectedEndTime.format(context);
```
Comment (bitfriend) [inline app/lib/features/events/widgets/change_date_sheet.dart:88]: I wonder why these statements were not wrapped by `setState()`.

**Pull request merged**